app/src/main/java/com/example/cr554/newsapp/MainActivity.java

```diff
@@ -93,12 +95,12 @@ private void redrawRecycler(){
         newsList.getAdapter().notifyDataSetChanged();
     }
 
-    private String getNewsUrl(){
+    private String getNewsUrl(String topic){
         String baseUri = getString(R.string.baseuri);
         Uri u = Uri.parse(baseUri).buildUpon()
-                .appendQueryParameter("api_key","e6678f97-7cd9-4156-88f7-f5f508c29ed3")
-                .appendQueryParameter("format", "json")
-                .appendQueryParameter("nojsoncallack","1")
+                .appendQueryParameter("show-fields","thumbnail")
+                .appendQueryParameter("q",topic)
+                .appendQueryParameter("api-key", APIKey.FLICKER_API_KEY)
```

Comment:
API keys are tricky, but I think the general consensus is that you should use them as "build properties". https://medium.com/@cassioso/a-strategy-to-secure-your-api-keys-using-gradle-b9c107272860#.ilte1o4ie would give a pretty good idea of what I mean :)